### DISCUSSION TITLE
bitmap  准确度问题

### DISCUSSION THREAD

**Discussion by xiaoL1211** [apache/doris#12331] [A - General / Q&A] [answered]
我设置的字段  count   bitmap bitmap_union
通过BITMAP_UNION_COUNT 查询
最后查询的数据和 实际插入的数据总是少一条。
查询 57146
实际插入 57147 

SQL ：insert into login_device_agg_day_org (`stat_time`,`org_id`,`count`)values ('2022-09-01',1,bitmap_hash('cbf7438e192011ec8d4f0242ac140002')),('2022-09-01',1,bitmap_hash('cbfa55e4192011ec8d4f0242ac140002')),('2022-09-01',1,bitmap_hash('cbf8bc09192011ec8d4f0242ac140002')),('2022-09-01',1,bitmap_hash('cc00a1c5192011ec8d4f0242ac140002')),('2022-09-01',1,bitmap_hash('cbff0847192011ec8d4f0242ac140002')),('2022-09-01',1,bitmap_hash('cbfc091d192011ec8d4f0242ac140002'))......

其中  bitmap_hash中的值唯一。stat_time，org_id一样

**Comment by LemonLiTree** [comment, marked as answer]
可能在做bitmap_hash时候出现了hash冲突，建议使用int32适用to_bitmap来做，如果表中没有合适int32字段，可以在考虑参考官网中全局字典的方式实现(https://doris.apache.org/zh-CN/docs/data-operate/import/import-way/spark-load-manual?_highlight=%E5%85%A8%E5%B1%80#%E5%85%A8%E5%B1%80%E5%AD%97%E5%85%B8)   或者   在做ETL时候通过一定算法代码加入

**Reply by xiaoL1211** [reply]
我把需要bitmap_hash的值进行了一次md5 问题就解决了

**Reply by xiaoL1211** [reply]
非常感谢

**Reply by xiaoL1211** [reply]
请问 bitmap_hash有64位的吗？32位还是谁冲突
